BostonData/lambda_function/get_alerts_intent.py

```diff
@@ -0,0 +1,99 @@
+"""
+
+City services in alerts dict:
+    Street Cleaning
+    Trash and recycling
+    City building hours
+    Parking meters
+    Tow lot
+
+Alerts for day are fetched from dictionary with city service as key
+
+
+Example:
+
+service_alerts['Street Cleaning'] = "Today is the third Tuesday of the month \\
+and street cleaning is running on a normal schedule."
+
+
+"""
+
+
+from alexa_utilities import build_response, build_speechlet_response
+from streetaddress import StreetAddressParser
+from bs4 import BeautifulSoup
+import requests
+import alexa_constants
+import urllib
+from enum import Enum
+from enum import auto
+
+class Services(Enum):
+    STREET_CLEANING = 'Street cleaning'
+    TRASH = 'Trash and recycling'
+    CITY_BUILDING_HOURS = 'City building hours'
+    PARKING_METERS = 'Parking meters'
+    TOW_LOT = 'Tow lot'
+    
+
+def get_alerts_info(intent, session):
+    """
+    Generate response object with information about citywide alerts
+    """
+    reprompt_text=None
+    print("IN GET_ALERTS_INFO, SESSION: " + str(session))
+    alerts = get_alerts()
+    print("DICTIONARY WITH ALERTS SCRAPED FROM BOSTON.GOV: " + str(alerts))
+    alerts = prune_normal_responses(service_alerts)
+    print("DICTIONARY AFTER PRUNING: " + str(alerts))
+    speech_output = alerts_to_speech_output(alerts)
+    session_attributes = session.get('attributes', {})
+    should_end_session = True   # leave this as True for right now
+    return build_response(session_attributes, build_speechlet_response(
+            intent['name'], speech_output, reprompt_text, should_end_session))
+
+
+def alerts_to_speech_output(alerts):
+    """
+    Return a string that contains all alerts or a message that city services are operating normally
+    """
+    if len(alerts) == 0:        # there are no alerts!
+        return "There are no alerts. City services are operating on their normal schedule"
+    else:
+        all_alerts = ""
+        for alert in alerts.values():
+            all_alerts += alert
+        return all_alerts
+        
+
+def prune_normal_responses(service_alerts):
+    """
+    Remove any text scraped from Boston.gov that aren't actually alerts.
+    For example, parking meters, city building hours, and trash and 
+    recycling are described "as on a normal schedule"
+    """
+
+    tow_lot_normal_message = "The tow lot is open from 7 a.m. - 11 p.m. "
+    tow_lot_normal_message += "Automated kiosks are available 24 hours a day, "
+    tow_lot_normal_message += "seven days a week for vehicle releases."
+
+    # for any defined service, if its alert is that it's running normally, 
+    # remove it from the dictionary
+    for service in Services:
+        if str.find(service_alerts[service.value], "normal") != -1: # this is a leap of faith
```

Comment:
This will throw an error if `service.value` is not a key in `service_alerts`. Lets try something like:
`if service.value in service_alerts and  str.find(service_alerts[service.value], "normal") != -1:`